0 Products/40 Zubax Babel/Tutorials/10 Developing custom applications.md

```diff
@@ -1,4 +1,263 @@
-# Developing custom applications
+# Developing custom applications #
+Ok, you have your Zubax babel and want it to do something special. First of all you will need babel firmware source code, which can be found [here](https://github.com/Zubax/zubax_babel)
+In order to build babel firmware *nix-environment will be needed.  If you use any kind of Linux - good for you, you probably already have everything needed(except for gnu arm toolchain). If (like me) you have only win machine, you will need:
 
-Coming soon!
+- [git](https://git-scm.com/download/win)
+- gnu utilities, which may be found [here](http://gnuwin32.sourceforge.net/)
+- [Python 3+](https://www.python.org/downloads/)
+- gnu arm toolchain. It is very version-dependant, so you should use [this version](https://launchpad.net/gcc-arm-embedded/4.9/4.9-2015-q3-update)
 
+After you download and install everything make sure you have all these your environment variable PATH set up correctly and have access to all necessary utilities from console. For example, my PATH variable looks like this:
+
+<img src="path.png" class="thumbnail" title="PATH">
+
+Now its time to try to build the original firmware. Go to console and clone the repository
+
+```git clone https://github.com/Zubax/zubax_babel```
```

Comment:
Are you sure it is necessary to put such a simple command on a separate line? Also, missing punctuation at the end of the previous line.